src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+            size++;
+        } else {
+            addLast(value);
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (isEmpty()) {
+            checkIndexWhenAdd(index);
+            head = tail = new Node<>(null, value, null);
+            size++;
+        } else if (index == 0) {
+            addFirst(value);
+            size++;
+        } else if (index == size) {
+            addLast(value);
+            size++;
+        } else {
+            checkIndexWhenAdd(index);
+            Node<T> node = node(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            node.prev.next = newNode;
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
This is correct, but I suggest to change it to for each to make it more readable

Reply:
agree